secure_fw/partitions/internal_trusted_storage/tfm_its_req_mngr.c

```diff
@@ -66,6 +67,27 @@ static psa_status_t tfm_its_set_req(const psa_msg_t *msg)
     size_remaining = msg->in_size[1];
     offset = 0;
 
+#ifdef TFM_ITS_ENCRYPTED
```

Comment:
This is a difficult to read with the ifdefs. I think that it can be simplified  to be more readable so I have a suggestion. Use/update it if you think that it is more clear for you as well.

```suggestion
#if defined(TFM_ITS_ENCRYPTED) &&  (PSA_FRAMEWORK_HAS_MM_IOVEC != 1)

bool do_size_check = true;

#ifdef TFM_PARTITION_PROTECTED_STORAGE
if (asset_info.client_id != TFM_SP_PS) {
     do_size_check = false;
}
#endif 

        if(do_size_check && (size_remaining > sizeof(asset_data))){
            return PSA_ERROR_BUFFER_TOO_SMALL;
        }
 #endif
```

Reply:
Ah I moved this out into its own function, hope that is easier to read